### PR TITLE
download models

### PR DESCRIPTION
**Pull request by eecaro**
Download models via the CLI:
- model name is required in order to download 
- they can also filter by tags and version 
- the user must be in the project directory if a project_token is not specified 
- if more than one model is found, user will be prompted to select a model to download
- model will be downloaded to specified output_path, otherwise it will be downloaded to the current directory under the filename `<version>_<model_name>.txt`
  - If they specify a dir only, the filename will also be `<version>_<model_name>.txt`

Docs: 
https://docs.metismachine.io/v1.2/docs/usage#section-models

### CODE REVIEW

File: src/models/models.cpp [hunk 1]
```diff
@@ -40,4 +28,141 @@ void Models::list(std::string project_token, std::map<std::string, std::string>
...
+            }
+          } else{
+            download_path = output_path + "/" + output_file;
+          }
+        }
+      } else {
+        download_path = output_path;
+      }
+    }
+    return download_path;
```
Comment (jeremytregunna) [inline src/models/models.cpp]: This whole function is hard to follow. Nested if statements inside nested if statements generally are not easy to follow. If it's easy to break this out to be easier to follow, please do so, otherwise it can pass for now.

**Review by jeremytregunna — CHANGES_REQUESTED** [review]

**Review by jeremytregunna — APPROVED** [review]

**Review by kmussel — APPROVED** [review]
